packages/expect/src/asymmetricMatchers.ts

```diff
@@ -163,40 +161,28 @@ class ObjectContaining extends AsymmetricMatcher<Record<string, unknown>> {
       );
     }
 
-    if (this.inverse) {
-      for (const property in this.sample) {
-        if (
-          hasProperty(other, property) &&
-          equals(this.sample[property], other[property]) &&
-          !emptyObject(this.sample[property]) &&
-          !emptyObject(other[property])
-        ) {
-          return false;
-        }
-      }
+    let result = true;
 
-      return true;
-    } else {
-      for (const property in this.sample) {
-        if (
-          typeof this.sample[property] === 'object' &&
-          !(this.sample[property] instanceof AsymmetricMatcher)
-        ) {
-          this.sample[property] = objectContaining(
-            this.sample[property] as Record<string, unknown>,
-          );
-        }
-
-        if (
-          !hasProperty(other, property) ||
-          !equals(this.sample[property], other[property])
-        ) {
-          return false;
-        }
+    for (const property in this.sample) {
+      if (
+        typeof this.sample[property] === 'object' &&
+        !(this.sample[property] instanceof AsymmetricMatcher)
+      ) {
+        this.sample[property] = objectContaining(
```

Comment:
this mutates like #10711 tries to avoid, no?

Reply:
It's the same line, and the other one can be landed for a patch while this needs to wait. This will then ofc need rebase on the version with the mutation removed